frontend/src/utilities/useWatchNotebook.tsx

```diff
@@ -42,9 +42,9 @@ export const useWatchNotebook = (
     const watchNotebook = (notebookName: string) => {
       getNotebook(projectName, notebookName)
         .then((data: Notebook) => {
+          setNotebook(data);
           setLoaded(true);
           setLoadError(undefined);
-          setNotebook(data);
```

Comment:
If i set the notebook after i set `loaded`, then the NotebookController page will get `loaded` first, but it cannot get `isNotebookRunning` because the notebook is not set, so it will go to the empty page, which is the spawner page. This is not React 18 so every setState will trigger a re-render.

Reply:
That sounds like your code should not have two states. If you're setting your loaded state based on the notebook state, you should just derive it from the notebook state.

Reply:
You are right... Why do I have two states here... 🤯